drivers/SmartThings/sonos/src/lunchbox/rest.lua

```diff
@@ -165,9 +157,18 @@ local function handle_response(sock)
 end
 
 local function execute_request(client, request, retry_fn)
-  if client.socket == nil then connect(client) end
+  if not client._active then
+    return nil, "Called `execute request` on a terminated REST Client"
+  end
 
-  local should_retry = retry_fn or function() return false end
+  if client.socket == nil then
+    local success, err = connect(client)
+    if not success then return nil, err end
+  end
+
+  if type(should_retry) ~= "function" then
```

Comment:
Looks like `should_retry` needs to be initalized somewhere, and I'm guessing this should be checking if `retry_fn()` is a function? I am somewhat confused how this wasn't caught in luacheck unless I'm looking at this wrong

Reply:
Good catch. Accidentally removed the local initializer during the translation.